### PR TITLE
Upgrade Files

### PR DESCRIPTION
**Pull request by NelkGames**
Random damage fixed, but I need more concrete confirmation
Flying using pokemon is working

Pokemon don't get exp in pokeplayer mode, but maybe I leave it as it is (it avoids abuse of players because the player is always faster than AI)

Bugs
(apparently he corrected himself ?) ~~The player's collision box is buggy when untransformed, but is corrected if using shift~~
-Player can't attack while in pokemob form (max one hit and then damage is disabled) using the pokemon by default the attack is normal

### CODE REVIEW

File: src/main/java/pokecube/pokeplayer/inventory/ContainerPokemob.java [hunk 1]
```diff
@@ -64,10 +65,10 @@ public boolean isItemValid(ItemStack stack)
             public ItemStack onTake(PlayerEntity playerIn, ItemStack stack)
             {
                 ItemStack old = getStack();
-//                if(FMLCommonHandler.instance().getEffectiveSide() == Dist.DEDICATED_SERVER)
-//                {
+                if(Dist.DEDICATED_SERVER != null)
```
Comment (Thutmose) [inline src/main/java/pokecube/pokeplayer/inventory/ContainerPokemob.java]: this is probably not what you want to do here, this will never be null.

File: src/main/java/pokecube/pokeplayer/inventory/ContainerPokemob.java [hunk 2]
```diff
@@ -78,10 +79,10 @@ public ItemStack onTake(PlayerEntity playerIn, ItemStack stack)
             public void putStack(ItemStack stack)
             {
                 super.putStack(stack);
-//                if(FMLCommonHandler.instance().getEffectiveSide() == Dist.DEDICATED_SERVER)
-//                {
+                if(Dist.DEDICATED_SERVER != null)
```
Comment (Thutmose) [inline src/main/java/pokecube/pokeplayer/inventory/ContainerPokemob.java]: Same here

File: src/main/java/pokecube/pokeplayer/network/handlers/AttackLocation.java [hunk 3]
```diff
@@ -61,15 +63,15 @@ public void handleCommand(IPokemob pokemob)
...
             defaults.readFromBuf(buffer);
             defaults.handleCommand(pokemob);
         }
     }
 
-    /*@Override
+    @Override
     public void writeToBuf(ByteBuf buf)
     {
         super.writeToBuf(buf);
```
Comment (Thutmose) [inline src/main/java/pokecube/pokeplayer/network/handlers/AttackLocation.java]: you need to write location to buf here as well, see AttackLocationHandler for how to do that

File: src/main/java/pokecube/pokeplayer/tileentity/TileEntityTransformer.java [hunk 4]
```diff
@@ -287,10 +264,11 @@ public void onWalkedOn(final Entity entityIn)
             poke.setPokemonNickname(tag.getString("oldName"));
             PokecubeCore.LOGGER.debug("Converting {} back to a human", player.getDisplayName().getString());
             tag.remove("oldName");
-            tag.remove("is_a_Player");
+//            tag.remove("is_a_player");
```
Comment (Thutmose) [inline src/main/java/pokecube/pokeplayer/tileentity/TileEntityTransformer.java]: unless you replaced where it adds this to flag as a player, this may cause some issues later if removed

Comment (NelkGames) [reply src/main/java/pokecube/pokeplayer/tileentity/TileEntityTransformer.java]: > unless you replaced where it adds this to flag as a player, this may cause some issues later if removed

here I removed that part and put info.detach();
fixed the problem where the pokecube got the block effect
